### PR TITLE
gh-81002: Add tests for termios

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
* Issue: gh-81002

### CODE REVIEW

File: Lib/test/test_termios.py [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+import unittest
+from test.support.import_helper import import_module
+
+termios = import_module('termios')
+
+
+class TestFunctions(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
```
Comment (sobolevn) [inline Lib/test/test_termios.py]: You use `setUp` in `test_tty`. Is it intentional?

**Review by sobolevn — COMMENTED** [review]
Thanks for working on this!

File: Lib/test/test_termios.py [hunk 2]
```diff
@@ -12,12 +12,10 @@ class TestFunctions(unittest.TestCase):
 
     @classmethod
     def setUpClass(cls):
-        try:
-            cls.stream = open('/dev/tty', 'wb', buffering=0)
-        except OSError:
-            raise unittest.SkipTest("Cannot open '/dev/tty'")
+        cls.master_fd, cls.fd = os.openpty()
```
Comment (sobolevn) [inline Lib/test/test_termios.py]: Some systems might not have it: `.. availability:: Unix, not Emscripten, not WASI.`

Other tests use something like `@unittest.skipUnless(hasattr(os, 'openpty'), 'need os.openpty()')`

File: Lib/test/test_termios.py (hunk 1 above)
Comment (serhiy-storchaka) [reply Lib/test/test_termios.py]: It has less tests. Actually, I am not sure what is better.

File: Lib/test/test_termios.py (hunk 2 above)
Comment (serhiy-storchaka) [reply Lib/test/test_termios.py]: Are there platforms that have termios, but not os.open? I'll add the guard, but first I want to run tests on buildbots to see what other issues can be here.

File: Lib/test/test_termios.py [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+
+    def test_tcsetattr_errors(self):
+        attrs = termios.tcgetattr(self.fd)
+        self.assertRaises(TypeError, termios.tcsetattr, self.fd, termios.TCSANOW, tuple(attrs))
+        self.assertRaises(TypeError, termios.tcsetattr, self.fd, termios.TCSANOW, attrs[:-1])
+        self.assertRaises(TypeError, termios.tcsetattr, self.fd, termios.TCSANOW, attrs + [0])
+        for i in range(6):
+            attrs2 = attrs[:]
+            attrs2[i] = 2**1000
+            self.assertRaises(OverflowError, termios.tcsetattr, self.fd, termios.TCSANOW, attrs2)
```
Comment (serhiy-storchaka) [inline Lib/test/test_termios.py:66]: This test crashed before #110366.

**Comment by serhiy-storchaka** [comment]
!buildbot AMD64 FreeBSD PR

File: Lib/test/test_termios.py (hunk 3 above)
Comment (serhiy-storchaka) [reply Lib/test/test_termios.py:66]: There is one such platform: WebAssembly.

File: Lib/test/test_termios.py (hunk 2 above)
Comment (serhiy-storchaka) [reply Lib/test/test_termios.py]: There is one such platform: WebAssembly.

**Pull request merged**